urllib3/response.py

```diff
@@ -351,3 +363,77 @@ def readinto(self, b):
         else:
             b[:len(temp)] = temp
             return len(temp)
+
+    def _read_chunked(self, amt):
+        def readline():
+            r = ""
+            while True:
+                ch = self._fp._safe_read(1)
+                r += ch
+                if r.endswith("\r\n"):
+                    return r
```

Comment:
Is this really needed? My suspicion is that httplib/http.client will provide a socket with readline already defined on it (by having previously called `socket.makefile` on it).